test/FunctionalCollectionTest.js

```diff
@@ -1,39 +1,107 @@
 import {FunctionalCollection} from './../src/FunctionalCollection';
-
 var assert = require('assert');
 
+var items = {
+    123: {
+        id: 123
+    },
+    125: {
+        id: 125
+    },
+    127: {
+        id: 127
+    }
+};
+
+/**
+ * Testing collection sample.
+ *
+ * @type {FunctionalCollection}
+ */
+var sampleCollection = new FunctionalCollection(() => {
```

Comment:
Every test must have complete control of the SUT, and therefore every test must create a new instance. You may, however, use a centralized factory for that.